group_project_v2/mixins.py

```diff
@@ -232,10 +246,58 @@ def studio_view(self, context):  # pylint: disable=unused-argument, no-self-use
         return fragment
 
 
-class DashboardMixin(object):
+class DashboardXBlockMixin(object):
+    """ Mixin for an XBlock that has dashboard views """
     def dashboard_view(self, context):
         raise NotImplementedError(MUST_BE_OVERRIDDEN)
 
+    def dashboard_detail_view(self, context):
+        raise NotImplementedError(MUST_BE_OVERRIDDEN)
```

Comment:
Is a details view always required for every class that uses `DashboardXBlockMixin`? If not, this should be a "Not available" error rather than a "subclasses MUST override this method" error.

Also, I think the distinction here is fuzzy. For the `GroupProjectXBlock`, the `dashboard_view` is an independent view that stands on its own. But for the stage XBlocks, the `dashboard_view` is meant to be embedded and the `dashboard_details_view` is/will be an independent view that stands on its own. It might be more consistent to make `dashboard_view` always a top-level view and have it render its children using a `dashboard_embed_view` or something? I'm not sure on this, but I wanted to mention it and see what you think.